src/Compilers/Test/Utilities/CSharp/BasicCompilationUtils.cs

```diff
@@ -37,7 +37,9 @@ private static VisualBasicCompilation CreateCompilationWithMscorlib(string sourc
             return VisualBasicCompilation.Create(assemblyName, new[] { tree }, references, options);
         }
 
-        private static BasicTestBase s_instance = new BasicTestBase();
+        private static BasicTestBase s_instance;
+
+        private static BasicTestBase Instance => s_instance ?? (s_instance = new BasicTestBase());
```

Comment:
This isn't what concerns me, InterlockedExchange wouldn't prevent the second invocation of the CTOR. It would prevent the property from returning different objects, which might be a source of intermittent and different to debug failures.